SMLHelper/Utility/PrefabUtils.cs

```diff
@@ -3,33 +3,114 @@
+        public static void AddBasicComponents(GameObject prefab, string classId, TechType techType, LargeWorldEntity.CellLevel cellLevel)
+        {
+            prefab.EnsureComponent<PrefabIdentifier>().ClassId = classId;
+            
+            if (techType != TechType.None)
+            {
+                prefab.EnsureComponent<TechTag>().type = techType;
+            }
+            
+            prefab.EnsureComponent<LargeWorldEntity>().cellLevel = cellLevel;
+            
+            var renderers = prefab.GetComponentsInChildren<Renderer>(true);
+            if (renderers != null)
+            {
+                prefab.EnsureComponent<SkyApplier>().renderers = renderers;
+            }
+        }
+
+        /// <summary>
+        /// Adds and configures the <see cref="Constructable"/> component on the specified prefab.
         /// </summary>
-        /// <param name="_object"></param>
-        /// <param name="classId"></param>
-        public static void AddBasicComponents(ref GameObject _object, string classId)
+        /// <param name="prefab">The prefab to operate on.</param>
+        /// <param name="techType">The tech type associated with the specified prefab.</param>
+        /// <param name="constructableFlags">A bitmask comprised of one or more <see cref="ConstructableFlags"/> that specify how the prefab should be treated during placement.</param>
+        public static void AddConstructable(GameObject prefab, TechType techType, ConstructableFlags constructableFlags)
         {
-            Rigidbody rb = _object.AddComponent<Rigidbody>();
-            _object.AddComponent<PrefabIdentifier>().ClassId = classId;
-            _object.AddComponent<LargeWorldEntity>().cellLevel = LargeWorldEntity.CellLevel.Near;
-            Renderer rend = _object.GetComponentInChildren<Renderer>();
-            rend.material.shader = Shader.Find("MarmosetUBER");
-            SkyApplier applier = _object.AddComponent<SkyApplier>();
-            applier.renderers = new Renderer[] { rend };
-            applier.anchorSky = Skies.Auto;
-            WorldForces forces = _object.AddComponent<WorldForces>();
-            forces.useRigidbody = rb;
+            var constructable = prefab.EnsureComponent<Constructable>();
+            constructable.controlModelState = true;
+            constructable.techType = techType;
+            constructable.allowedInBase = constructableFlags.HasFlag(ConstructableFlags.Base);
+            constructable.allowedInSub = constructableFlags.HasFlag(ConstructableFlags.Submarine);
+            constructable.allowedOutside = constructableFlags.HasFlag(ConstructableFlags.Outside);
+            constructable.allowedOnCeiling = constructableFlags.HasFlag(ConstructableFlags.Ceiling);
+            constructable.allowedOnConstructables = constructableFlags.HasFlag(ConstructableFlags.AllowedOnConstructable);
+            constructable.allowedOnWall = constructableFlags.HasFlag(ConstructableFlags.Wall);
         }
```

Comment:
Verify the techtype is not None before adding component and log and error instead if it is.